packages/image_picker/image_picker_platform_interface/lib/src/platform_interface/image_picker_platform.dart

```diff
@@ -305,4 +303,75 @@ abstract class ImagePickerPlatform extends PlatformInterface {
     );
     return pickedImages ?? <XFile>[];
   }
+
+  /// Returns true if the implementation supports [source].
+  ///
+  /// Defaults to true for the original image sources, `gallery` and `camera`,
+  /// for backwards compatibility.
+  bool supportsImageSource(ImageSource source) {
+    return source == ImageSource.gallery || source == ImageSource.camera;
+  }
+}
+
+/// A base class for an [ImagePickerPlatform] implementation that does not
+/// directly support [ImageSource.camera], but supports delegating to a
+/// provided [ImagePickerCameraDelegate].
+abstract class CameraDelegatingImagePickerPlatform extends ImagePickerPlatform {
+  /// A delegate to respond to calls that use [ImageSource.camera].
+  ///
+  /// When it is null, attempting to use [ImageSource.camera] will throw a
+  /// [StateError].
+  ImagePickerCameraDelegate? cameraDelegate;
+
+  @override
+  bool supportsImageSource(ImageSource source) {
+    if (source == ImageSource.camera) {
+      return cameraDelegate != null;
+    }
+    return super.supportsImageSource(source);
+  }
+
+  @override
+  Future<XFile?> getImageFromSource({
+    required ImageSource source,
+    ImagePickerOptions options = const ImagePickerOptions(),
+  }) async {
+    if (source == ImageSource.camera) {
+      final ImagePickerCameraDelegate? delegate = cameraDelegate;
```

Comment:
Is this necessary?

Reply:
Because Dart won't promote nullable fields (because in general it can't guarantee that you aren't running this code in the context of a subclass that overrode the field with a getter that does something crazy like return null every other time you call it), you have two options for cases like this:
- Make a local variable, null check it to promote it, and then use it as non-nullable.
- Null-check the field, then use `!` every time you access it.

I strongly prefer the former since it relies on the compiler to ensure that things are still right any time someone changes the logic, while the latter relies on humans to ensure it.